pkg/workflows/ring/plugin.go

```diff
@@ -0,0 +1,282 @@
+		"batchSize", cfg.BatchSize,
+		"timeToSync", cfg.TimeToSync,
+	)
+
+	return &Plugin{
+		store:         store,
+		arbiterScaler: arbiterScaler,
+		config:        config,
+		lggr:          logger.Named(lggr, "RingPlugin"),
+		batchSize:     cfg.BatchSize,
+		timeToSync:    cfg.TimeToSync,
+	}, nil
+}
+
+//coverage:ignore
+func (p *Plugin) Query(_ context.Context, _ ocr3types.OutcomeContext) (types.Query, error) {
+	return nil, nil
+}
+
+func (p *Plugin) Observation(ctx context.Context, _ ocr3types.OutcomeContext, _ types.Query) (types.Observation, error) {
+	var wantShards uint32
+	shardStatus := make(map[uint32]*pb.ShardStatus)
+
+	if p.arbiterScaler != nil {
+		status, err := p.arbiterScaler.Status(ctx, &emptypb.Empty{})
+		if err != nil {
+			p.lggr.Warnw("failed to get arbiter scaler status", "error", err)
+		} else {
+			wantShards = status.WantShards
+			shardStatus = status.Status
+		}
+	} else {
+		// Fallback to store if no arbiter scaler configured
+		shardHealth := p.store.GetShardHealth()
+		for shardID, healthy := range shardHealth {
+			shardStatus[shardID] = &pb.ShardStatus{IsHealthy: healthy}
+		}
+	}
+
+	allWorkflowIDs := make([]string, 0)
+	for wfID := range p.store.GetAllRoutingState() {
+		allWorkflowIDs = append(allWorkflowIDs, wfID)
+	}
+
+	pendingAllocs := p.store.GetPendingAllocations()
+	allWorkflowIDs = append(allWorkflowIDs, pendingAllocs...)
+
+	allWorkflowIDs = uniqueSorted(allWorkflowIDs)
+
+	observation := &pb.Observation{
+		ShardStatus: shardStatus,
+		WorkflowIds: allWorkflowIDs,
+		Now:         timestamppb.Now(),
+		WantShards:  wantShards,
+	}
+
+	return proto.MarshalOptions{Deterministic: true}.Marshal(observation)
+}
+
+//coverage:ignore
+func (p *Plugin) ValidateObservation(_ context.Context, _ ocr3types.OutcomeContext, _ types.Query, _ types.AttributedObservation) error {
+	return nil
+}
+
+func (p *Plugin) ObservationQuorum(_ context.Context, _ ocr3types.OutcomeContext, _ types.Query, aos []types.AttributedObservation) (quorumReached bool, err error) {
+	return quorumhelper.ObservationCountReachesObservationQuorum(quorumhelper.QuorumTwoFPlusOne, p.config.N, p.config.F, aos), nil
+}
+
+func (p *Plugin) collectShardInfo(aos []types.AttributedObservation) (shardHealth map[uint32]int, workflows []string, timestamps []time.Time, wantShardVotes []uint32) {
+	shardHealth = make(map[uint32]int)
+	for _, ao := range aos {
+		observation := &pb.Observation{}
+		if err := proto.Unmarshal(ao.Observation, observation); err != nil {
+			p.lggr.Warnf("failed to unmarshal observation: %v", err)
+			continue
+		}
+
+		for shardID, status := range observation.ShardStatus {
+			if status != nil && status.IsHealthy {
+				shardHealth[shardID]++
+			}
+		}
+
+		workflows = append(workflows, observation.WorkflowIds...)
+
+		if observation.Now != nil {
+			timestamps = append(timestamps, observation.Now.AsTime())
+		}
+
+		if observation.WantShards > 0 {
+			wantShardVotes = append(wantShardVotes, observation.WantShards)
```

Comment:
nit: It would be nice to have a map from node ID to its wantShards, for debugability.